src/components/admin/form-components/withSearchBar.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import { cleanupFalcorKeys } from 'lib/falcor/falcor-utilities';
+import { withFalcor } from 'components/hocs/falcor-hocs';
+import { debounce } from 'lib/utilities';
+import { getDisplayName } from 'lib/higher-order-helpers';
+
+export const withSearchableConcept = (
+  fields,
+  formatter,
+  category,
+) => WrappedField => {
+  const debounceTime = 250;
+  class Searchable extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        suggestions: [],
+      };
+
+      this.handleChange = this.handleChange.bind(this);
+      this.handleClick = this.handleClick.bind(this);
+
+      this.debounceSuggestions = debounce(query => {
+        if (!query.trim()) {
+          this.setState({ suggestions: [] });
+          return;
+        }
+
+        let pathSets = this.props.extraPathSets || [];
+        pathSets = pathSets.reduce(
+          (arr, pathSet) => {
+            arr.push(
+              ['search', category, query, { length: this.props.length }].concat(
+                pathSet,
+              ),
+            );
+            return arr;
+          },
+          [
+            [
+              'search',
+              category,
+              query,
+              { length: this.props.length },
+              _.uniq([...fields, ...this.props.fields, 'id']),
+            ],
+          ],
+        );
+
+        this.props.falcor.get(...pathSets).then(x => {
+          if (!x) {
+            this.setState({ suggestions: [] });
+            return;
+          }
+          x = cleanupFalcorKeys(x); // eslint-disable-line no-param-reassign
```

Comment:
I would prefer if we could remove these as we see them, and start actually dealing with what falcor is actually giving us. I've been contemplating writing a small falcor utilitity function named something like `falcorDataToArray` which takes a falcor pseudo array and returns an array. It would simply be something like `falcor.keys(psuedoArray).map(key => pseudoArray[key])`